src/codegen/dyngen.rs

```diff
@@ -5,6 +5,11 @@ use proc_macro2::Ident;
 /// Used to build the output tokens for dynamic bindings.
 #[derive(Default)]
 pub struct DynamicItems {
+    /// Tracks whether or not we contain any required symbols.
+    /// If so, the signature of the generated `from_library` function
+    /// will be altered to return a `Result<Self, ::libloading::Error>`
+    has_required: bool,
+
```

Comment:
So right now this is an all-or-nothing thing, right? Did you introduce this thinking in adding more granularity in the future, or can this be simplified?

Reply:
Indeed - but this bool can actually be eliminated now.
I had it here previously so we could avoid altering the signature of `from_library` - but my implementation for that was messy so I just opted to alter the signature.